Mlem/App/Globals/Definitions/HapticManager/HapticPriority.swift

```diff
@@ -0,0 +1,38 @@
+//
+//  HapticPriority.swift
+//  Mlem
+//
+//  Created by Eric Andrews on 2024-06-10.
+//
+
+import Foundation
+
+/// Enum to denote priority of haptic feedback. The levels denote:
+/// - sentinel: denotes that the user has selected no haptics. Should never be used as a haptic priority!
+/// - high: denotes a haptic that always plays if the user has selected any degree of haptic feedback
+/// - low: denotes a haptic that only plays if the user has selected "all" haptic feedback
+enum HapticPriority: String, CaseIterable, Comparable {
+    case sentinel, high, low
+    
+    var intValue: Int {
+        switch self {
+        case .sentinel: return 0
```

Comment:
With regards to this comment:

`/// - sentinel: denotes that the user has selected no haptics. Should never be used as a haptic priority!`

Maybe we could remove the `.sentinel` case from this and just store `HapticPriority?` for the setting, where `nil` replaces `.sentinel`.

Possibly OOS, since this is all copy-paste

Reply:
Opened an [issue](https://github.com/mlemgroup/mlem/issues/1106) for it